test/requests.js

```diff
@@ -703,6 +705,38 @@ var tests = function(web3) {
       });
     });
 
+    // NB: relies on the previous test setting value to 25 and the contract deployment setting
+    // original value to 5. `contractCreationBlockNumber` is set in the first test of this
+    // describe block.
+    it("should be able to read data via a call at a specified block (eth_call)", function(done){
+      var starting_block_number = null;
+      var call_data = contract.call_data;
+      call_data.to = contractAddress;
+
+      // TODO: Removing this callback hell would be nice.
```

Comment:
Callback hell can be somewhat removed now - web3 1.0 returns PromiEvents!

Also I wonder if we're not leaning too heavily on `requests.js` for our testing... I almost feel like this whole file should be broken out by method name.

Again, nothing that you need to change here, just idle musing.

Reply:
Actually, ideally we'd test setting a specific block number, as well as `initial`, `latest`, and `pending`... If you're up for it, perhaps you could create `test/call.js` if it doesn't already exist, and do that there?

Reply:
Definitely - that's a good idea.

Reply:
@benjamincburns Close to pushing in the revisions - would you like to move everything `eth_call` (and maybe even `estimateGas`?) into `test/call.js`? The file currently exists but only contains tests for an edge case where running a call could exceed the gasLimit.

Reply:
@benjamincburns Actually - restructuring the `requests.js` is likely a good idea but WDYT about doing it in a separate PR? Have a stack of PRs headed your way this week and maybe that one would be a good way to cap them off.